### PR TITLE
Docs: resolve warnings from sphinx build output

### PR DESCRIPTION
**Pull request by AJcodes**
The following PR addresses the warnings (and some errors) from the sphinx build output:
- Duplicate target names
  - Some links were edited to have unique names according to the context
  - Some links were fixed to ensure the same target
  - In the case of the links under the tensorflow-compatibility.rst page, the following [solution](https://www.tarantool.io/en/doc/latest/contributing/docs/sphinx-warnings/#duplicate-explicit-target-name) was used to ensure visual consistency rather than applying a unique name
- `:orphan:` was added to files that are not listed within the main toctree, but are accessible via other pages
- Some errors were produced due to labels being placed outside `{grid-item-card}`; those labels have been moved into the cards
- In `docs/conceptual/ai-pytorch-inception.md`, there were some labels that were causing warnings (e.g. `[înception_arch]`), I'm not sure if they were meant to link somewhere, but even on the official docs page they weren't being displayed properly. I've removed them but would appreciate feedback

Some trimming was applied to the files that were changed.

### CODE REVIEW

File: docs/conceptual/ai-pytorch-inception.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ The following sections contain case studies for the Inception V3 model.
 
 ### Inception V3 with PyTorch
 
-Convolution Neural Networks are forms of artificial neural networks commonly used for image processing. One of the core layers of such a network is the convolutional layer, which convolves the input with a weight tensor and passes the result to the next layer. Inception V3[^inception_arch] is an architectural development over the ImageNet competition-winning entry, AlexNet, using more profound and broader networks while attempting to meet computational and memory budgets.
```
Comment (neon60) [inline docs/conceptual/ai-pytorch-inception.md:56]: @peterjunpark I do not know, what were the purpose of these?

[^inception_arch] 
[^Stanford_deep_learning]
[^cross_entropy]

**Comment by neon60** [comment]
@peterjunpark @AJcodes are fixing warnings and errors of the ROCm documentation and some of the warnings coming from pages, which you and I edited in the past. Are you agreeing with those changes?

File: docs/conceptual/ai-pytorch-inception.md (hunk 1 above)
Comment (peterjunpark) [reply docs/conceptual/ai-pytorch-inception.md:56]: These are supposed to be footnote references. It looks like the footnotes themselves were removed in this commit (https://github.com/ROCm/ROCm/commit/890c735f53e2e51f72b16ddd75136b86083985bd#diff-2d54c599bf3456e76c0e6af531d0cfc2e9ce13faf3ff999f8243a0c51e67388aL1274), but not the anchors.

Comment (AJcodes) [reply docs/conceptual/ai-pytorch-inception.md:56]: > These are supposed to be footnote references. It looks like the footnotes themselves were removed in this commit ([890c735#diff-2d54c599bf3456e76c0e6af531d0cfc2e9ce13faf3ff999f8243a0c51e67388aL1274](https://github.com/ROCm/ROCm/commit/890c735f53e2e51f72b16ddd75136b86083985bd#diff-2d54c599bf3456e76c0e6af531d0cfc2e9ce13faf3ff999f8243a0c51e67388aL1274)), but not the references.

Would it make sense to add the relevant footnotes back in? Or is it best to just remove the references?

**Review by peterjunpark — APPROVED** [review]

File: docs/compatibility/ml-compatibility/jax-compatibility.rst [hunk 2]
```diff
@@ -22,14 +24,14 @@ support:
...
-  - ROCm JAX repository: `<https://github.com/ROCm/jax>`__
+  - ROCm JAX repository: `ROCm/jax <https://github.com/ROCm/jax>`_
 
   - See the :doc:`ROCm JAX installation guide <rocm-install-on-linux:install/3rd-party/jax-install>`
     to get started.
 
 - Official JAX release:
 
-  - Official JAX repository: `<https://github.com/jax-ml/jax>`__
+  - Official JAX repository: `jax-ml/jax <https://github.com/jax-ml/jax>`_
```
[the remark this answers is not in the document]
Comment (AJcodes) [reply docs/compatibility/ml-compatibility/jax-compatibility.rst:34]: Same, I was worried adding a target name would remove the icon, but fortunately it keeps it

**Comment by peterjunpark** [comment]
> @peterjunpark @AJcodes are fixing warnings and errors of the ROCm documentation and some of the warnings coming from pages, which you and I edited in the past. Are you agreeing with those changes?

Yeah, looks good to me @neon60. Thanks for the fixes @AJcodes.

File: docs/conceptual/ai-pytorch-inception.md (hunk 1 above)
Comment (peterjunpark) [reply docs/conceptual/ai-pytorch-inception.md:56]: I think the PyTorch footnotes could be helpful.

The [^Stanford_deep_learning] reference that points to some online curriculum doesn't seem very helpful to me though.

Comment (AJcodes) [reply docs/conceptual/ai-pytorch-inception.md:56]: @peterjunpark As @neon60 already merged the PR, we'll discuss maybe doing this in a future PR. But thanks for the input
